hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/launcher/ContainerLaunch.java

```diff
@@ -1539,34 +1547,17 @@ public void listDebugInformation(Path output) throws IOException {
       if (envVal == null || envVal.isEmpty()) {
         return Collections.emptySet();
       }
+      if (pattern == null) {
+        pattern = Pattern.compile("%(.*?)%");
```

Comment:
I think this could be extracted as well.

Reply:
Done.